File: Domotización - Interfaces/Domotización - Interfaces/PuertaManual.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domotización___Interfaces
{
    internal class PuertaManual : Puerta
    {
        public PuertaManual(int numPuerta) : base(numPuerta)
        {
        }

        public override void bloquear()
        {
            Bloqueada = true;
            Console.WriteLine($"Puerta número {NumPuerta} desbloqueada manualmente.");
        }

        public override void desbloquear()
        {
             Bloqueada = false;
            Console.WriteLine($"Puerta número {NumPuerta} desbloqueada manualmente.");
        }
    }
}

File: DomotizacionInterfaces/src/main/java/org/example/PuertaManual.java
package org.example;

public class PuertaManual extends Puerta {


    protected PuertaManual(int numPuerta) {
        super(numPuerta);
    }

    public void bloquear()
    {
        setBloqueada(true);
        System.out.println("Puerta número " + getNumPuerta() + " bloqueada manualmente");
    }

    public void desbloquear()
    {
        setBloqueada(false);
        System.out.println("Puerta número " + getNumPuerta() + " desbloqueada manualmente");
    }


}
